src/nbl/video/CVKSwapchain.cpp

```diff
@@ -3,49 +3,49 @@
 #include "nbl/video/CVKLogicalDevice.h"
 #include "nbl/video/surface/ISurfaceVK.h"
 #include "nbl/video/CVulkanImage.h"
+#include "nbl/video/CVulkanConnection.h"
 
-namespace nbl {
-namespace video
+namespace nbl::video
 {
 
-CVKSwapchain::CVKSwapchain(SCreationParams&& params, CVKLogicalDevice* dev) : ISwapchain(std::move(params)), m_device(dev)
+CVKSwapchain::CVKSwapchain(SCreationParams&& params, CVKLogicalDevice* dev)
+    : ISwapchain(dev, std::move(params)), m_device(dev)
 {
-    VkSwapchainCreateInfoKHR ci;
-    ci.sType = VK_STRUCTURE_TYPE_SWAPCHAIN_CREATE_INFO_KHR;
-    ci.pNext = nullptr;
-    ci.minImageCount = params.minImageCount;
-    ci.clipped = VK_TRUE;
-    ci.imageArrayLayers = params.arrayLayers;
-    // TODO function mapping ISurface::SFormat -> VkColorSpaceKHR
-    //ci.imageColorSpace = ...
-    ci.presentMode = static_cast<VkPresentModeKHR>(params.presentMode);
-    ci.imageExtent = { params.width, params.height };
-    ci.imageSharingMode = VK_SHARING_MODE_EXCLUSIVE; // single queue famility at a time
-    ci.queueFamilyIndexCount = params.queueFamilyIndices->size();
-    ci.pQueueFamilyIndices = params.queueFamilyIndices->data();
-    ci.preTransform = VK_SURFACE_TRANSFORM_IDENTITY_BIT_KHR;
-    ci.oldSwapchain = VK_NULL_HANDLE;
-    //ci.imageUsage = ... // TODO (we dont have the enum yet)
-    ci.flags = 0;
-    ci.compositeAlpha = VK_COMPOSITE_ALPHA_OPAQUE_BIT_KHR;
-    ci.imageFormat = static_cast<VkFormat>(params.surfaceFormat.format);
-    ci.surface = static_cast<ISurfaceVK*>(params.surface.get())->getInternalObject();
-
-    auto* vk = m_device->getFunctionTable();
-    VkDevice vkdev = m_device->getInternalObject();
-
-    VkSwapchainKHR sc;
-    vk->vk.vkCreateSwapchainKHR(vkdev, &ci, nullptr, &sc);
-
-    m_swapchain = sc;
+    VkSwapchainCreateInfoKHR createInfo = { VK_STRUCTURE_TYPE_SWAPCHAIN_CREATE_INFO_KHR };
+    createInfo.surface = static_cast<ISurfaceVK*>(m_params.surface.get())->getInternalObject();
+    createInfo.minImageCount = m_params.minImageCount;
+    createInfo.imageFormat = ISurfaceVK::getVkFormat(m_params.surfaceFormat.format);
+    createInfo.imageColorSpace = ISurfaceVK::getVkColorSpaceKHR(m_params.surfaceFormat.colorSpace);
+    createInfo.imageExtent = { m_params.width, m_params.height };
+    createInfo.imageArrayLayers = m_params.arrayLayers;
+
+    // Todo(achal): Probably need make an enum for this
```

Comment:
yes please

Reply:
also take it as an argument

Reply:
@Crisspl TODO (you can assign to someone) do the validation of image and buffer usages in OpenGL (can be asserts or something similar)

Reply:
I guess the file is now `CVulkanSwapchain` ?

Reply:
No, there's no `CVulkanSwapchain` file. AFAICR `CVKLogicalDevice` and `CVKSwapchain` are the only different ones.

Reply:
No, there's no `CVulkanSwapchain` file. AFAICR `CVKLogicalDevice` and `CVKSwapchain` are the only different ones.

Reply:
keep namign consistency, cause I have no idea what to review

Reply:
resolved